pkg/bee/api/api.go

```diff
@@ -215,12 +237,11 @@ func (c *Client) requestWithHeader(ctx context.Context, method, path string, hea
 		return err
 	}
 
-	if err = responseErrorHandler(r); err != nil {
-		return err
-	}
-
 	if v != nil && strings.Contains(r.Header.Get("Content-Type"), "application/json") {
 		_ = json.NewDecoder(r.Body).Decode(&v)
```

Comment:
Should we use err = json.NewDecoder(r.Body).Decode(&v) and return if not nil , else we do the `for` and return nil , right?

Reply:
this is done in responseErrorHandler